internal/proxy/proxy.go

```diff
@@ -734,7 +741,47 @@ func newSocketMount(ctx context.Context, conf *Config, pc *portConfig, inst Inst
 		}
 
 		address = net.JoinHostPort(a, fmt.Sprint(np))
-	} else {
+
+	case inst.UnixSocketPath != "" && strings.HasPrefix(version, "POSTGRES"):
```

Comment:
This case statement and the one below both duplicate code in the default path. How can we avoid that while keeping the intention clear?

Reply:
As one option, leaving the if/else in place makes it clear that we're doing either TCP or Unix. Then we could introduce a function e.g., `address, err := unixSocketPath(dir, version, address, inst.Name)` that give you more room to gracefully handle all the variation.

Reply:
Good idea. Done.